public/src/managers/ServerProvider.ts

```diff
@@ -375,23 +374,20 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
 
     public async updateSubmission(courseID: number, submissionID: number, approve: boolean): Promise<boolean> {
         const result = await this.grpcHelper.updateSubmission(courseID, submissionID, approve);
-        if (result.status.getCode() !== 0) {
-            return false;
-        }
-        return true;
+        return this.responseCodeSuccess(result);
     }
 
     public async rebuildSubmission(assignmentID: number, submissionID: number): Promise<ISubmission | null> {
         const result = await this.grpcHelper.rebuildSubmission(assignmentID, submissionID);
-        if (result.status.getCode() !== 0 || !result.data) {
+        if (!this.responseCodeSuccess(result) || !result.data) {
```

Comment:
With an updated method that also checks `result.data`, this pattern becomes:
```typescript
if (this.responseSuccess(result)) {
    return this.toISubmission(result.data);
}
return null;
```
(and similar for almost all the uses above.)